src/rust/compress/src/main.rs

```diff
@@ -0,0 +1,31 @@
+use std::env;
+use std::fmt::Write;
+use std::iter;
+use std::process::exit;
+
+fn compress(input: &str) -> String {
+    let mut chars = input.chars().peekable();
+    let mut output = String::new();
+
+    while let Some(ch) = chars.next() {
+        let count = iter::from_fn(|| chars.next_if_eq(&ch)).count();
+        write!(output, "{}{}", count + 1, ch).ok();
+    }
+
+    output
+}
+
+fn main() {
+    match env::args().skip(1).next() {
+        Some(input) => println!("{}", compress(&input)),
+        None => {
+            eprintln!("No input provided.");
+            exit(1);
```

Comment:
I would consider using `panic!` macro instead of these 2 lines.

Reply:
I'd rather just remove this as it's not really relevant.